main.sh

```diff
@@ -47,6 +47,7 @@ fi
 __dir="$(cd "$(dirname "${BASH_SOURCE[${__b3bp_tmp_source_idx:-0}]}")" && pwd)"
 __file="${__dir}/$(basename "${BASH_SOURCE[${__b3bp_tmp_source_idx:-0}]}")"
 __base="$(basename "${__file}" .sh)"
+__origin="$(cd "$(dirname "$(readlink -f "${BASH_SOURCE[${__b3bp_tmp_source_idx:-0}]}")")" && pwd)"
```

Comment:
I think one problem with this is that readlink isn't ubiquitously available like this https://stackoverflow.com/questions/1055671/how-can-i-get-the-behavior-of-gnus-readlink-f-on-a-mac

Reply:
Thank you @kvz for pointing that out, and unwillingly throwing me into a rabbit hole with it 😉.
It seems with this restriction a one-liner solution cannot be achived. Could you imagine introducing a function like `__readlink_f` for this functionality?

I challenged myself to write an equivalent using `cd`, `pwd -P` and `readlink`, which should be available on all platforms. I got a lot right, but I'm struggling with relative symlinks. The function will not be very big, but ~100 lines it will be I guess.

Reply:
I have been down that rabbit hole before, and i think it is a case where we should admit defeat unfortunately :pensive:

Reply:
I made it out the 🐇🕳️ again. I'm not sure how deep I went, but I would love to here what you think, if you find the time: [rdlink.md](https://github.com/kvz/bash3boilerplate/files/8325074/rdlink.md) (rename it back to sh)

I ran a lot of tests but only debian based newer OSs and only bash 3.2.57 and bash 5.